File: nysol_python/mcmd/methods/convTex/db/smp/mnrjoin.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mnrjoin'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
date,price
20080123,10
20080123,20
20080203,10
20080203,35
200804l0,50
'''
db['idatas'].append(data)

data={}
data['name']='ref1.csv'
data['text']='''
date,priceF,priceT,avg
20080203,5,15,150
20080203,40,50,200
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
日付項目の値が ``20080203`` で、 ``amount`` 項目の値が ``5`` 以上 ``15`` 未満の入力データ行には ``avg=150`` を、
``40`` 以上 ``50`` 未満の行には ``avg=200`` を結合する。
'''
script['sh_code']='''
mnrjoin k=date f=avg m=ref1.csv R=priceF,priceT r=price%n i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mnrjoin(k="date", f="avg", m="ref1.csv", R="priceF,priceT", rf="price%n", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='未結合データ出力'
script['comment']='''
``n=True`` を指定することで、参照ファイルにマッチしない入力ファイルの行( ``avg=`` がNULL値の行)も出力し、
``N=True`` を指定することで、入力ファイルにマッチしない参照ファイルの行( ``price=`` がNULL値の行)も出力する。
いわゆる外部結合である。
'''
script['sh_code']='''
mnrjoin k=date f=avg m=ref1.csv R=priceF,priceT r=price%n -n -N i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mnrjoin(k="date", f="avg", m="ref1.csv", R="priceF,priceT", rf="price%n", n=True, N=True, i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mnrjoin.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
date,price
20080123,10
20080123,20
20080203,10
20080203,35
200804l0,50
EOF
)}

File.open("ref1.csv","w"){|fpw| fpw.write(
<<'EOF'
date,priceF,priceT,avg
20080203,5,15,150
20080203,40,50,200
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
For records where the value of date field is \verb|20080203|, select those records in the input data where \verb|amount| field is more than \verb|5| but less than \verb|15| and join field where \verb|avg=150|. For records where \verb|amount| field is more than \verb|40| but less than \verb|50|, join field \verb|avg=200|.
EOF
scp=<<'EOF'
more dat1.csv
more ref1.csv
mnrjoin k=date f=avg m=ref1.csv R=priceF,priceT r=price%n i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Output unmatched data"
comment=<<'EOF'
Use \verb|-n| to return all records in the input data even if they do not match with those in the reference file (row where \verb|avg=| Null), and use \verb|-N| to return records in the reference file even if they do not match with those in the input file (rows where \verb|price=| null). This is known as outer-join.
EOF
scp=<<'EOF'
mnrjoin k=date f=avg m=ref1.csv R=priceF,priceT r=price%n -n -N i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)
